test/liquid-vault.js

```diff
@@ -378,16 +378,88 @@ describe('LiquidVault', function () {
       new Date().getTime() + 3000, // uint deadline
     )).to.emit(uniswapPair, 'Burn')
 
-    infinityBalanceAfter = await infinity.balanceOf(owner.address)
-    wethBalanceAfter = await weth.balanceOf(owner.address)
+    infinityBalanceAfter = await infinity.balanceOf(owner.address);
+    wethBalanceAfter = await weth.balanceOf(owner.address);
 
-    assert.isTrue(wethBalanceAfter.gt(wethBalanceBefore))
-    assert.isTrue(infinityBalanceAfter.gt(infinityBalanceBefore))
+    assert.isTrue(wethBalanceAfter.gt(wethBalanceBefore));
+    assert.isTrue(infinityBalanceAfter.gt(infinityBalanceBefore));
 
     const lpBalanceAfterRemoveLiquidity = await uniswapPair.balanceOf(owner.address);
 
-    assertBNequal(lpBalanceAfterRemoveLiquidity, 0)
+    assertBNequal(lpBalanceAfterRemoveLiUserquidity);
+  });
+
+  it('should be able to claim 1 batch after 1 purchase with 0% fees then transfer lptokens to another user and remove eth liquidity from uniswap', async function() {
+    const purchaseValue = utils.parseEther('1');
+    const transferToLiquidVault = utils.parseUnits('20000', baseUnit); // 20.000 tokens
+
+    await infinity.transfer(liquidVault.address, transferToLiquidVault);
+    assertBNequal(await infinity.balanceOf(liquidVault.address), transferToLiquidVault);
+    
+    await liquidVault.purchaseLP({ value: purchaseValue });
+    const lockedLP = await liquidVault.getLockedLP(owner.address, 0);
+    const { donationShare } = await liquidVault.config();
+    const stakeDuration = await liquidVault.getStakeDuration();
+    const lpBalanceBeforeClaim = await uniswapPair.balanceOf(owner.address);
+
+    await ganache.setTime((bn(lockedLP[2]).add(stakeDuration)).toNumber());
+    const claimLP = await liquidVault.claimLP();
+    const receipt = await claimLP.wait();
+
+    const { holder, amount, exitFee, claimed } = receipt.events[0].args;
+    const estimatedFeeAmount = lockedLP[1].mul(donationShare).div(bn('100'));
+    const lpBalanceAfterClaim = await uniswapPair.balanceOf(owner.address);
+
+    assert.strictEqual(holder, owner.address);
+    assert.isTrue(claimed);
+    assertBNequal(amount, lockedLP[1]);
+    assertBNequal(exitFee, estimatedFeeAmount);
+    assertBNequal(amount.sub(exitFee), lpBalanceAfterClaim.sub(lpBalanceBeforeClaim));
+
+    uniswapPairUserBalanceBefore = await uniswapPair.balanceOf(user.address);
+    uniswapPairOwnerBalanceBefore = await uniswapPair.balanceOf(owner.address);
+
+    await uniswapPair.transfer(user.address, lpBalanceAfterClaim);
+
+    uniswapPairUserBalanceAfter = await uniswapPair.balanceOf(user.address);
```

Comment:
lines 419-410, 424-425.  It's better to hardcode LP balance values in test to be able to compare hardcoded LP balances for user, owner before transfer on line 422 and after

Reply:
resolved